android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -887,6 +887,34 @@ class GeckoEngineSession(
         })
     }
 
+    /**
+     * See [EngineSession.sendPlacementAttributionEvent]
+     */
+    override fun sendPlacementAttributionEvent(
+        aid: String,
+        onResult: (Boolean) -> Unit,
+        onException: (Throwable) -> Unit,
+    ) {
+        geckoSession.sendPlacementAttributionEvent(aid).then({
+                response ->
+            val errorMessage = "Invalid value: unable to send placement attribution event through Gecko Engine."
+            if (response == null) {
+                logger.error(errorMessage)
+                onException(
+                    java.lang.IllegalStateException(errorMessage),
+                )
+                return@then GeckoResult()
+            }
+            onResult(response)
+            GeckoResult<Boolean>()
+        }, {
+                throwable ->
```

Comment:
this line seems unformatted:)

Reply:
Good point, I reformatted similar functions so it's more readable.